### DISCUSSION TITLE
Motivation behind deprecations

### DISCUSSION THREAD

**Discussion by Riduidel** [structurizr/java#204] [General]
Hello, in the latest versions, some concepts have been deprecated or removed that I tend to use ... a lot ... in Aadarchi.

Namely

* `CodeElement` has been removed in [`v1.21.0`](https://github.com/structurizr/java/releases/tag/v1.21.0), which is frankly a bad decision from my point of view
* `Enterprise` has been deprecated in [`v1.23.0`](https://github.com/structurizr/java/releases/tag/v1.23.0)
* `Location` has been deprecated in the same version

Could it be possible to discuss why these elements disappear (ideally before they disappear) ?

**Comment by simonbrowndotje** [comment]
- Code elements: these are not used by any of the core Structurizr tooling (the "component finders" haven't been supported for a few years now, and the Structurizr UI doesn't support rendering them), so I took the decision to remove them.
- Enterprise boundary and location: these concepts are too limited, so I recently added support for nested groups.

**Reply by Riduidel** [reply]
I understand the rationale behind CodeElement deletion, but as I'm using them (for various reasons), I'm a bit puzzled on how to emulate that feature (excepted by abusing the element properties)

**Reply by simonbrowndotje** [reply]
If I ever resurrect the component finder code, I'd probably just make my own `CodeElement` class in that library, along with a way to store them, perhaps something like `Map<Component,Set<CodeElement>>`.

**Reply by Riduidel** [reply]
Would you be interested if someone (like me) took care of that part ? Maybe I could try some kind of responsibility delegation.

**Reply by simonbrowndotje** [reply]
You are more than welcome to fork the component finder code and take it into your own repo ... I have enough other code to support. 😀

**Reply by Riduidel** [reply]
Isn't it already in structurizr extensions ? (where I'm a contributor)

**Comment by simonbrowndotje** [comment]
Closing this discussion in favour of https://github.com/structurizr/java-extensions/discussions/69.
